plugins/swagger/swagger.go

```diff
@@ -0,0 +1,180 @@
+func (s *swaggerInstaller) Install(builder service.Builder, cfg *nirvana.Config) error {
+	var err error
+	wrapper(cfg, func(c *config) {
+		for _, v := range c.config {
+			path, filePath, err := splitConfig(v)
+			if err != nil {
+				panic(err)
+			}
+			_, err = pathExists(filePath)
+			if err != nil {
+				panic(err)
+			}
+
+			err = builder.AddDescriptor(definition.Descriptor{
+				Path:     path,
+				Consumes: []string{definition.MIMEAll},
+				Produces: []string{definition.MIMEAll},
+				Definitions: []definition.Definition{{
+					Method:  definition.Get,
+					Results: definition.DataErrorResults(""),
+					Function: func(ctx context.Context) ([]byte, error) {
+						b, err := ioutil.ReadFile(filePath)
+						if err != nil {
+							return nil, err
+						}
+						return b, nil
+					},
+				}},
+			})
+		}
+	})
+	return err
+}
+
+// Uninstall uninstalls stuffs after server terminating.
+func (s *swaggerInstaller) Uninstall(builder service.Builder, cfg *nirvana.Config) error {
+	return nil
+}
+
+// Disable returns a configurer to disable current plugin for a certain nirvana server.
+func Disable() nirvana.Configurer {
+	return func(c *nirvana.Config) error {
+		// Set to nil will delete plugin config from nirvana config.
+		c.Set(ExternalConfigName, nil)
+		return nil
+	}
+}
+
+type config struct {
+	config []string
+}
+
+// SetOptions returns a configurer to set swagger options.
+func SetOptions(option Option) nirvana.Configurer {
+	return func(c *nirvana.Config) error {
+		wrapper(c, func(c *config) {
+			c.config = option.Config
+		})
+		return nil
+	}
+}
+
+func wrapper(c *nirvana.Config, f func(c *config)) {
+	conf := c.Config(ExternalConfigName)
+	var cfg *config
+	if conf == nil {
+		// Default config.
+		cfg = &config{}
+	} else {
+		// Panic if config type is wrong.
+		cfg = conf.(*config)
+	}
+	f(cfg)
+	c.Set(ExternalConfigName, cfg)
+}
+
+// Option contains basic configurations of swagger.
+type Option struct {
+	Config []string `desc:"Colon separated of apiVersion:swaggerFilePath"`
+}
+
+// Name returns plugin name.
+func (p *Option) Name() string {
+	return ExternalConfigName
+}
+
+// NewDefaultOption creates default option.
+func NewDefaultOption() *Option {
+	return &Option{}
+}
+
+// NewOption creates option with config.
+func NewOption(config []string) *Option {
+	return &Option{
+		Config: config,
+	}
+}
+
+// Configure configures nirvana config via current options.
+func (p *Option) Configure(cfg *nirvana.Config) error {
+	cfg.Configure(
+		SetOptions(*p),
+	)
+	return nil
+}
+
+func splitConfig(config string) (org, name string, err error) {
+	if !strings.Contains(config, ":") {
+		return "", "", fmt.Errorf("Please specify the swagger config with apiVersion:swaggerFilePath")
+	}
+
+	set := strings.Split(config, ":")
+
+	if len(set) != 2 {
+		return "", "", fmt.Errorf("Please specify the repository name with apiVersion:swaggerFilePath")
+	}
+
+	path := fmt.Sprintf("/apis/%s/swagger.json", strings.TrimSpace(set[0]))
```

Comment:
Don't assume that users will use `/apis/`.

Reply:
updated